File: remote-continuous-query/src/test/scala/org/littlewings/infinispan/remotecq/Book.scala
package org.littlewings.infinispan.remotecq

import org.infinispan.protostream.annotations.{ProtoField, ProtoMessage}
import org.infinispan.protostream.descriptors.Type

object Book {
  def apply(isbn: String, title: String, price: Int): Book = {
    val book = new Book

    book.isbn = isbn
    book.title = title
    book.price = price

    book
  }
}

@ProtoMessage(name = "Book")
class Book extends Serializable {
  private[remotecq] var isbn: String = _
  private[remotecq] var title: String = _
  private[remotecq] var price: Int = _

  @ProtoField(number = 1, name = "isbn", required = true, `type` = Type.STRING)
  def getIsbn: String = isbn

  def setIsbn(isbn: String): Unit = this.isbn = isbn

  @ProtoField(number = 2, name = "title", required = true, `type` = Type.STRING)
  def getTitle: String = title

  def setTitle(title: String): Unit = this.title = title

  @ProtoField(number = 3, name = "price", required = true, defaultValue = "0", `type` = Type.INT32)
  def getPrice: Int = price

  def setPrice(price: Int): Unit = this.price = price
}

File: remote-opentelemetry-tracing/src/main/java/org/littlewings/infinispan/remote/otlp/Book.java
package org.littlewings.infinispan.remote.otlp;

import org.infinispan.protostream.annotations.ProtoField;
import org.infinispan.protostream.descriptors.Type;

public class Book {
    @ProtoField(number = 1, type = Type.STRING)
    String isbn;

    @ProtoField(number = 2, type = Type.STRING)
    String title;

    @ProtoField(number = 3, type = Type.INT32, defaultValue = "0")
    int price;

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }
}
